neo/rawio/openephysbinaryrawio.py

```diff
@@ -253,30 +252,50 @@ def _parse_header(self):
                     # 'states' was introduced in OpenEphys v0.6. For previous versions, events used 'channel_states'
                     if "states" in info or "channel_states" in info:
                         states = info["channel_states"] if "channel_states" in info else info["states"]
+
                         if states.size > 0:
                             timestamps = info["timestamps"]
                             labels = info["labels"]
-                            rising = np.where(states > 0)[0]
-                            falling = np.where(states < 0)[0]
 
-                            # infer durations
+                            # Identify unique channels based on state values
+                            channels = np.unique(np.abs(states))
+
+                            rising_indices = []
+                            falling_indices = []
+
+                            for channel in channels:
+                                # Find rising and falling edges for each channel
```

Comment:
Could we get a slightly more explanatory comment (maybe above L266). Based on the code it seems like the rising are encode by +1 channel so if I'm at channel 1 then +1 is rising and -1 is falling and then for channel 2 it would be +2 and -2.  What would the array of states actually look like? So I'm imagining something like:

```
all channels are packed into the same `states` array. So the states array includes positive and negative values for each channel for example channel one rising would be +1 and channel one falling would be -1. So we can check for where each channel appears to get each TTL rise and fall for each channel
```

Or something like this. Since we have to support multiple versions it is helpful in the cases where the format is changed for us to know exactly how the former format was set up.

Reply:
Ok, should have been fixed now!